Source/Core/Core/PowerPC/Interpreter/Interpreter_SystemRegisters.cpp

```diff
@@ -549,34 +549,12 @@ void Interpreter::isync(UGeckoInstruction inst)
 void Interpreter::mcrfs(UGeckoInstruction inst)
 {
   UpdateFPSCR(&FPSCR);
-  u32 fpflags = ((FPSCR.Hex >> (4 * (7 - inst.CRFS))) & 0xF);
-  switch (inst.CRFS)
-  {
-  case 0:
-    FPSCR.FX = 0;
-    FPSCR.OX = 0;
-    break;
-  case 1:
-    FPSCR.UX = 0;
-    FPSCR.ZX = 0;
-    FPSCR.XX = 0;
-    FPSCR.VXSNAN = 0;
-    break;
-  case 2:
-    FPSCR.VXISI = 0;
-    FPSCR.VXIDI = 0;
-    FPSCR.VXZDZ = 0;
-    FPSCR.VXIMZ = 0;
-    break;
-  case 3:
-    FPSCR.VXVC = 0;
-    break;
-  case 5:
-    FPSCR.VXSOFT = 0;
-    FPSCR.VXSQRT = 0;
-    FPSCR.VXCVI = 0;
-    break;
-  }
+  const u32 shift = 4 * (7 - inst.CRFS);
+  const u32 fpflags = (FPSCR.Hex >> shift) & 0xF;
+
+  // If any exception bits were read, clear them
+  FPSCR.Hex &= ~((0xF << shift) & (FPSCR_FX | FPSCR_ANY_X));
```

Comment:
Could you explain why the mask is AND'd with `FPSCR_FX | FPSCR_ANY_X`?

Reply:
If we read a bit that represents whether a certain exception has happened, we need to clear that bit. But if we read some other bit, like the rounding mode, FPSCR, or whether a certain exception should be enabled, that bit shouldn't be cleared.